app/lib/storage/utils.ts

```diff
@@ -0,0 +1,25 @@
+import * as mkdirp from "mkdirp"
+import ErrnoException = NodeJS.ErrnoException
+
+/**
+ * The Utils module contains storage-related helper functions
+ */
+
+/**
+ * Checks asynchronously if a certain directory exist. If it does not, creates
+ * the directory and all missing parents in the path. This is equivalent to
+ * `mkdir -p` in POSIX.
+ * @param {string} path
+ * @returns {Promise<any>}
+ */
+export const ensurePath = async (path: string): Promise<any> => {
```

Comment:
Promise<any> could be Promise<boolean> as resolve logic just returns true

Reply:
Yes, I thought of the same, actually even `boolean` is too broad since the function either fails or returns `true`, so what we really need is a [unit type](https://en.wikipedia.org/wiki/Unit_type), the closest thing in TypeScript would be `void` so the end result would be:
```typescript
export const ensurePath = async (path: string): Promise<void> => {
  return new Promise((resolve, reject) => {
    mkdirp(path, (error: ErrnoException, made: mkdirp.Made) => {
      if (error) {
        reject(error)
      } else {
        resolve()
      }
    })
  })
} 
```